Titulos API: Agregar capa de repositorios

francosang opened the pull request:
Agrega archivo de repositorio con todas las funciones de base de datos.
Reemplaza funciones basadas en callbacks por funciones basadas en promises y async/await.

francosang committed:
Titulos API: agregar capa de repositorios

zaknein reviewed (commented):
Van a querer borrar títulos? Yo pensé que siempre quedarían ahí nomás. 
Después entiendo cómo funciona todo. Menos el tomodel y pool

francosang committed:
Rename toModel

francosang commented:
@zaknein 

> Van a querer borrar títulos? Yo pensé que siempre quedarían ahí nomás. 

En un principio, mientras los titulos no son automaticos ni programables, vamos a tener la posibilidad de agregarlos y borrarlos manualmente.

> Después entiendo cómo funciona todo. Menos el tomodel y pool

Con respecto al `pool`:
 - Es el nombre que tiene la conexión a la base de datos.
 - Cuando uso la librería para conectarme a la base, ese es el nombre que tiene.
 - Como no sabía que ponerle, lo dejé asi 😅 .
 - Despues lo cambio.

Con respecto al `toModel`:
 - Generalmente las aplicaciones tienen distintos tipos de clases/objetos para las mismas entidades.
 - Por ejemplo, si tu aplicación trabaja con Productos, es normal tener diferentes clases para distinas formas de representar el producto.
 - La clase `ProductoDatabase` para representar los campos que tiene el producto en la base de datos.
 - O la clase `ProductoApi` para representar los campos del producto que se trabajan a nivel de la API.
 - Pueden haber mas clases, para la logica de negocio u otro tipo de usos.
 - La teoria es que en algunas partes de la aplicacion tu producto puede estar formado por campos diferentes, entonces hay que crear objetos o clases diferentes.

En definitiva, esa funcion `toModel` la cree con ese proposito. Separar la representación que tiene el titulo en la base de datos de la representación del titulo en el resto de la aplicación.

Pero ahora que me preguntas esto, me doy cuenta que no tiene sentido. 
Mejor mantenerlo sencillo.
Así que acabo de renombrarlo a algo mas claro y generico como `convertRowToObject`.

En definitiva, tenia este bloque de codigo repetido tres veces...
```
const result = {
    id: row.id,
    titulo: row.name,
    created: row.created,
    updated: row.updated,
  }
```
...asi que cree un funcion para tenerlo solo una vez y llamar a la misma funcion.
Reutilizar codigo y evitar repeticion son los principios mas importares 🤣

zaknein commented:
Ahi va, buenísimo
